File: Test Files/operators.swift
//
// Copyright 2018 Vinícius Jorge Vendramini
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

let x = 0
let y = x + 1
let z = 2 + 3

print(x)
print(y)
print(z)

//
let multiplication = 0 * 1
let division = 10 / 2
let subtraction = 3 - 1
let sum = 4 + 1

print(multiplication)
print(division)
print(subtraction)
print(sum)

// Without spaces
let multiplication2 = 0*1
let division2 = 10/2
let subtraction2 = 3-1
let sum2 = 4+1

print(multiplication2)
print(division2)
print(subtraction2)
print(sum2)

// With negative numbers
var multiplication3 = 0 * -1
var division3 = -10 / 2
var subtraction3 = -3 - 1
var sum3 = 4 + -1

print(multiplication3)
print(division3)
print(subtraction3)
print(sum3)

// In assignments (as opposed to variable declarations)
multiplication3 = 1 * 2
division3 = 1 / 2
subtraction3 = 1 - 2
sum3 = 1 + 2

print(multiplication3)
print(division3)
print(subtraction3)
print(sum3)

// Equality
if x == 0 {
	print("true")
}

if x == 1 {
	print("false")
}

if x != 1 {
	print("true")
}

File: Test Files/operators.kt
//
// Copyright 2018 Vinicius Jorge Vendramini
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

fun main(args: Array<String>) {
	val x: Int = 0
	val y: Int = x + 1
	val z: Int = 2 + 3

	println(x)
	println(y)
	println(z)

	val multiplication: Int = 0 * 1
	val division: Int = 10 / 2
	val subtraction: Int = 3 - 1
	val sum: Int = 4 + 1

	println(multiplication)
	println(division)
	println(subtraction)
	println(sum)

	val multiplication2: Int = 0 * 1
	val division2: Int = 10 / 2
	val subtraction2: Int = 3 - 1
	val sum2: Int = 4 + 1

	println(multiplication2)
	println(division2)
	println(subtraction2)
	println(sum2)

	var multiplication3: Int = 0 * -1
	var division3: Int = -10 / 2
	var subtraction3: Int = -3 - 1
	var sum3: Int = 4 + -1

	println(multiplication3)
	println(division3)
	println(subtraction3)
	println(sum3)

	multiplication3 = 1 * 2
	division3 = 1 / 2
	subtraction3 = 1 - 2
	sum3 = 1 + 2

	println(multiplication3)
	println(division3)
	println(subtraction3)
	println(sum3)

	if (x == 0) {
		println("true")
	}

	if (x == 1) {
		println("false")
	}

	if (x != 1) {
		println("true")
	}
}
